api/src/test/kotlin/no/nav/helsearbeidsgiver/inntektsmelding/api/trenger/TrengerRouteKtTest.kt

```diff
@@ -33,34 +35,44 @@ import no.nav.helsearbeidsgiver.inntektsmelding.api.utils.ApiTest
 import no.nav.helsearbeidsgiver.inntektsmelding.api.validation.ValidationResponse
 import no.nav.helsearbeidsgiver.utils.json.fromJson
 import no.nav.helsearbeidsgiver.utils.json.toJson
-import no.nav.helsearbeidsgiver.utils.json.toJsonStr
 import no.nav.helsearbeidsgiver.utils.test.date.april
 import no.nav.helsearbeidsgiver.utils.test.date.desember
 import no.nav.helsearbeidsgiver.utils.test.date.februar
 import no.nav.helsearbeidsgiver.utils.test.date.januar
 import no.nav.helsearbeidsgiver.utils.test.date.mai
 import no.nav.helsearbeidsgiver.utils.test.date.mars
 import no.nav.helsearbeidsgiver.utils.test.json.removeJsonWhitespace
+import no.nav.helsearbeidsgiver.utils.test.mock.mockConstructor
 import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Assertions.assertNull
+import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
 import java.util.UUID
 
 private const val PATH = Routes.PREFIX + Routes.TRENGER
 
 class TrengerRouteKtTest : ApiTest() {
 
+    @BeforeEach
+    fun setup() {
+        clearAllMocks()
+    }
+
     @Test
     fun `skal returnere resultat og status CREATED når trenger virker`() = testApi {
+        val mockClientId = UUID.randomUUID()
+        val expectedJson = Mock.trengerResponseJson()
+
         mockTilgang(Tilgang.HAR_TILGANG)
 
-        coEvery {
-            anyConstructed<RedisPoller>().getString(any(), any(), any())
-        } returns Mock.trengerDataOk.toJsonStr(TrengerData.serializer())
+        coEvery { anyConstructed<RedisPoller>().hent(mockClientId, any(), any()) } returns Mock.trengerDataOkJson
 
-        val expectedJson = Mock.trengerResponseJson()
+        val response = mockConstructor(TrengerProducer::class) {
+            every { anyConstructed<TrengerProducer>().publish(any(), any()) } returns mockClientId
```

Comment:
Må legge til mocking av `TrengerProducer` her og i andre tester fordi vi går fra å bruke `RedisPoller.getString` til `RedisPoller.hent`. Det er nødvendig fordi `RedisPoller.hent` også brukes av tilgangssjekken (som vi ikke vil mocke her), som gjør at vi ikke kan bruke `any()` for alle argumentene.